letsencrypt/acme/jose.py

```diff
@@ -0,0 +1,100 @@
+"""JOSE."""
+import base64
+import binascii
+
+import Crypto.PublicKey.RSA
+
+from letsencrypt.acme import util
+
+
+def _leading_zeros(arg):
+    if len(arg) % 2:
+        return '0' + arg
+    return arg
+
+
+class JWK(util.JSONDeSerializable, util.ImmutableMap):
+    # pylint: disable=too-few-public-methods
+    """JSON Web Key.
+
+    .. todo:: Currently works for RSA public keys only.
+
+    """
+    __slots__ = ('key',)
+    schema = util.load_schema('jwk')
+
+    @classmethod
+    def _encode_param(cls, param):
+        """Encode numeric key parameter."""
+        return b64encode(binascii.unhexlify(
+            _leading_zeros(hex(param)[2:].rstrip('L'))))
+
+    @classmethod
+    def _decode_param(cls, param):
+        """Decode numeric key parameter."""
+        return long(binascii.hexlify(b64decode(param)), 16)
+
+    def to_json(self):
+        """Serialize to JSON."""
+        return {
+            'kty': 'RSA',  # TODO
+            'n': self._encode_param(self.key.n),
+            'e': self._encode_param(self.key.e),
+        }
+
+    @classmethod
+    def _from_valid_json(cls, jobj):
```

Comment:
:) I was suprised you responded so quickly... but after thinking about it a bit more I understood.  I should have edited it with an - I understand comment instead of deleteing... I will replace the original comment above so the history makes sense.